app/src/main/java/com/android/brewr/ui/overview/Overview.kt

```diff
@@ -36,6 +51,46 @@ fun OverviewScreen(
 ) {
   // State to track whether we're in "Gallery" or "Explore" mode
   var currentSection by remember { mutableStateOf("Gallery") }
+  var coffeeShops by remember { mutableStateOf<List<Coffee>>(emptyList()) }
+
+  val coroutineScope = rememberCoroutineScope()
+  val context = LocalContext.current
+  var permissionGranted by remember { mutableStateOf(false) }
+
+  val locationPermissionLauncher =
+      rememberLauncherForActivityResult(
+          contract = ActivityResultContracts.RequestMultiplePermissions(),
+          onResult = { permissions ->
+            permissionGranted =
+                permissions[Manifest.permission.ACCESS_FINE_LOCATION] == true ||
+                    permissions[Manifest.permission.ACCESS_COARSE_LOCATION] == true
+          })
+
+  LaunchedEffect(Unit) {
+    permissionGranted =
+        ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) ==
+            PackageManager.PERMISSION_GRANTED ||
+            ContextCompat.checkSelfPermission(
+                context, Manifest.permission.ACCESS_COARSE_LOCATION) ==
+                PackageManager.PERMISSION_GRANTED
+
+    if (!permissionGranted) {
+      locationPermissionLauncher.launch(
+          arrayOf(
+              Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION))
+    }
+  }
+
+  LaunchedEffect(permissionGranted) {
```

Comment:
This is a very good idea, it will be simpler and less annoying for users to ask permission in one time on the overviewscreen rather than ask permission for every screen